### PR TITLE
rv32i: try a different section identifier

### PR DESCRIPTION
**Pull request by ppannuto**
Also shorten the name becuase of Mach-O's 16 character limit

### Pull Request Overview

This hopefully fixes the issue introduced in #1778 by using a different identifier. With an underscore instead of a comma, it appears to get linked into the `.text` section again:

```
$ readelf.py -S target/riscv32imac-unknown-none-elf/release/hifive1.elf
There are 22 section headers, starting at offset 0x18a568

Section Headers:
  [Nr] Name              Type            Addr     Off    Size   ES Flg Lk Inf Al
  [ 0]                   NULL            00000000 000000 000000 00      0   0  0
  [ 1] .stack            NOBITS          80000000 0000f4 000800 00  WA  0   0  1
  [ 2] .text             PROGBITS        20400000 000200 00c61c 00 AXM  0   0 16
  [ 3] .storage          PROGBITS        2040c61c 00c81c 0001e4 00  AX  0   0  1
  [ 4] .apps             PROGBITS        20430000 00ca00 000000 00  AX  0   0  1
  [ 5] .relocate         PROGBITS        80000800 00ca00 0011f0 00  WA  0   0  4
  [ 6] .sram             NOBITS          800019f0 00dbf0 001e10 00  WA  0   0  4
  [ 7] .debug_str        PROGBITS        00000000 00dbf0 049434 01  MS  0   0  1
  [ 8] .debug_loc        PROGBITS        00000000 057024 01ddad 00      0   0  1
  [ 9] .debug_abbrev     PROGBITS        00000000 074dd1 000be7 00      0   0  1
  [10] .debug_info       PROGBITS        00000000 0759b8 06e411 00      0   0  1
  [11] .debug_aranges    PROGBITS        00000000 0e3dc9 001468 00      0   0  1
  [12] .debug_ranges     PROGBITS        00000000 0e5231 009170 00      0   0  1
  [13] .debug_macinfo    PROGBITS        00000000 0ee3a1 000002 00      0   0  1
  [14] .debug_pubnames   PROGBITS        00000000 0ee3a3 01c65e 00      0   0  1
  [15] .debug_pubtypes   PROGBITS        00000000 10aa01 02b984 00      0   0  1
  [16] .debug_frame      PROGBITS        00000000 136388 003f48 00      0   0  4
  [17] .debug_line       PROGBITS        00000000 13a2d0 0169b7 00      0   0  1
  [18] .comment          PROGBITS        00000000 150c87 000012 01  MS  0   0  1
  [19] .symtab           SYMTAB          00000000 150c9c 02f0f0 10     21 12018  4
  [20] .shstrtab         STRTAB          00000000 17fd8c 0000e5 00      0   0  1
  [21] .strtab           STRTAB          00000000 17fe71 00a6f4 00      0   0  1
```

### Testing Strategy

Compiling.

### TODO or Help Wanted

@alistair23 does this fix?

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make formatall`.

### CODE REVIEW

**Comment by alistair23** [comment]
This doesn't fix the OpenTitan regression

**Comment by alistair23** [comment]
It looks like the `riscv,trap_vecs` is still included when it wasn't before

```
Sections:
Idx Name          Size      VMA       LMA       File off  Algn
  0 .flash_header 00000004  20000000  20000000  00000200  2**0
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
  1 .stack        00001000  10000000  10000000  00000204  2**0
                  ALLOC
  2 .text         0000de6c  20000010  20000010  00000210  2**4
                  CONTENTS, ALLOC, LOAD, READONLY, CODE
  3 .riscv,trap_vecs 00000088  2000de7c  2000de7c  0000e07c  2**1
                  CONTENTS, ALLOC, LOAD, READONLY, CODE
  4 .storage      000000fc  2000df04  2000df04  0000e104  2**0
                  CONTENTS, ALLOC, LOAD, READONLY, CODE
  5 .apps         00000001  20030000  20030000  0000e200  2**0
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
  6 .relocate     00001028  10001000  2000e000  0000e400  2**2
                  CONTENTS, ALLOC, LOAD, DATA
  7 .sram         00004bd8  10002028  10002028  0000f428  2**2
                  ALLOC
  8 .debug_str    0004dcb0  00000000  00000000  0000f428  2**0
                  CONTENTS, READONLY, DEBUGGING, OCTETS
  9 .debug_loc    00024391  00000000  00000000  0005d0d8  2**0
                  CONTENTS, READONLY, DEBUGGING, OCTETS
 10 .debug_abbrev 00000bf4  00000000  00000000  00081469  2**0
                  CONTENTS, READONLY, DEBUGGING, OCTETS
 11 .debug_info   0007a1d6  00000000  00000000  0008205d  2**0
                  CONTENTS, READONLY, DEBUGGING, OCTETS
 12 .debug_aranges 000015f0  00000000  00000000  000fc233  2**0
                  CONTENTS, READONLY, DEBUGGING, OCTETS
 13 .debug_ranges 00009bb0  00000000  00000000  000fd823  2**0
                  CONTENTS, READONLY, DEBUGGING, OCTETS
 14 .debug_macinfo 00000002  00000000  00000000  001073d3  2**0
                  CONTENTS, READONLY, DEBUGGING, OCTETS
 15 .debug_pubnames 0001d5e5  00000000  00000000  001073d5  2**0
                  CONTENTS, READONLY, DEBUGGING, OCTETS
 16 .debug_pubtypes 0002e031  00000000  00000000  001249ba  2**0
                  CONTENTS, READONLY, DEBUGGING, OCTETS
 17 .debug_frame  000042cc  00000000  00000000  001529ec  2**2
                  CONTENTS, READONLY, DEBUGGING, OCTETS
 18 .debug_line   00017f32  00000000  00000000  00156cb8  2**0
                  CONTENTS, READONLY, DEBUGGING, OCTETS
 19 .comment      00000012  00000000  00000000  0016ebea  2**0
                  CONTENTS, READONLY
SYMBOL TABLE:
```

**Comment by alistair23** [comment]
This diff fixes the problem:

```diff
diff --git a/chips/ibex/src/chip.rs b/chips/ibex/src/chip.rs
index 5c4c06895..b16c1d589 100644
--- a/chips/ibex/src/chip.rs
+++ b/chips/ibex/src/chip.rs
@@ -210,7 +210,7 @@ pub unsafe fn configure_trap_handler() {
         .write(mtvec::trap_addr.val(_start_trap_vectored as u32 >> 2) + mtvec::mode::Vectored)
 }
 
-#[link_section = ".riscv,trap_vecs"]
+#[link_section = ".riscv_trap_vecs"]
 #[export_name = "_start_trap_vectored"]
 #[naked]
 pub extern "C" fn _start_trap_vectored() -> ! {
```

File: chips/ibex/src/chip.rs [hunk 1]
```diff
@@ -210,7 +210,7 @@ pub unsafe fn configure_trap_handler() {
         .write(mtvec::trap_addr.val(_start_trap_vectored as u32 >> 2) + mtvec::mode::Vectored)
 }
 
-#[link_section = ".riscv,trap_vectored"]
+#[link_section = ".riscv,trap_vecs"]
```
Comment (ppannuto) [inline chips/ibex/src/chip.rs:213]: Ugh, that's what I get for going quickly; I missed this one, which is the critical one, and once this is no longer a comma, we're back to the original problem from #1778.

This is going to take a little digging I think; and may fundamentally require a `cfg` given the first 5 solutions I found online.

**Comment by ppannuto** [comment]
Unfortunately that diff reintroduces the original problem. Ultimately the underlying issue is that when tests run for a crate, they're built for the host platform and executed there. So if there are custom section names, they must be valid on the host platform. Because the world has to be difficult, Mach-O _requires_ commas and ELF _disallows_ commas, so there is literally no section name that is valid on both OS X and Unix-like hosts environments, which means we _must_ `cfg` around this.

This was already being done for this exact problem in the rv32i arch crate, this now replicates the solution for the ibex chip.

**Comment by bradjc** [comment]
So this is good to go then?

**Comment by alistair23** [comment]
I have been testing with it, so I hope so

**Pull request merged**